src/js/schemas/service-schema/EnvironmentVariables.js

```diff
@@ -21,12 +21,20 @@ let EnvironmentVariables = {
         if (variableMap == null) {
           return [];
         }
-        return Object.keys(variableMap).map(function (key) {
-          return Hooks.applyFilter('variablesGetter', {
+
+        return Object.keys(variableMap).reduce(function (memo, key) {
+          if ((key == null || key === 'undefined')
```

Comment:
right here it's checking if the key is a string `undefined`. i forget the reason for this, but i remember sometime before this point, `undefined` got converted to `'undefined'`